include/irr/asset/IVirtualTexture.h

```diff
@@ -0,0 +1,750 @@
+            bnd.stageFlags = asset::ISpecializedShader::ESS_FRAGMENT;
+            bnd.type = asset::EDT_COMBINED_IMAGE_SAMPLER;
+            bnd.samplers = nullptr; //samplers are left for user to specify at will
+        };
+
+        fillBinding(bindings[0], _pgtBinding, 1u);
+
+        uint32_t i = 1u;
+        if (getFloatViews().size())
+        {
+            fillBinding(bindings[i], _fsamplersBinding, getFloatViews().size());
+            ++i;
+        }
+        if (getIntViews().size())
+        {
+            fillBinding(bindings[i], _isamplersBinding, getIntViews().size());
+            ++i;
+        }
+        if (getUintViews().size())
+        {
+            fillBinding(bindings[i], _usamplersBinding, getUintViews().size());
+        }
+
+        return retval;
+    }
+    template <typename DS_t>
+    std::pair<core::smart_refctd_dynamic_array<typename DS_t::SWriteDescriptorSet>, core::smart_refctd_dynamic_array<typename DS_t::SDescriptorInfo>>
+        getDescriptorSetWrites_internal(DS_t* _dstSet, uint32_t _pgtBinding = 0u, uint32_t _fsamplersBinding = 1u, uint32_t _isamplersBinding = 2u, uint32_t _usamplersBinding = 3u) const
+    {
+        using writes_t = core::smart_refctd_dynamic_array<typename DS_t::SWriteDescriptorSet>;
+        using info_t = core::smart_refctd_dynamic_array<typename DS_t::SDescriptorInfo>;
+        using retval_t = std::pair<writes_t, info_t>;
+
+        const uint32_t writeCount = 1u+(getFloatViews().size()?1u:0u)+(getIntViews().size()?1u:0u)+(getUintViews().size()?1u:0u);
+        const uint32_t infoCount = 1u + getFloatViews().size() + getIntViews().size() + getUintViews().size();
+
+        auto writes_array = core::make_refctd_dynamic_array<writes_t>(writeCount);
+        auto* writes = writes_array->data();
+        auto info_array = core::make_refctd_dynamic_array<info_t>(infoCount);
+        auto* info = info_array->data();
+
+        writes[0].binding = _pgtBinding;
+        writes[0].arrayElement = 0u;
+        writes[0].count = 1u;
+        writes[0].descriptorType = EDT_COMBINED_IMAGE_SAMPLER;
+        writes[0].dstSet = _dstSet;
+        writes[0].info = info;
+        info[0].desc = createPageTableView();
+        info[0].image.imageLayout = EIL_UNDEFINED;
+        info[0].image.sampler = nullptr; //samplers are left for user to specify at will
+
+        uint32_t i = 1u, j = 1u;
+        if (getFloatViews().size())
+        {
+            writes[i].binding = _fsamplersBinding;
+            writes[i].arrayElement = 0u;
+            writes[i].count = getFloatViews().size();
+            writes[i].descriptorType = EDT_COMBINED_IMAGE_SAMPLER;
+            writes[i].dstSet = _dstSet;
+            writes[i].info = info+j;
+            for (uint32_t j0 = j; (j-j0)<writes[i].count; ++j)
+            {
+                info[j].desc = getFloatViews().begin()[j-j0];
+                info[j].image.imageLayout = EIL_UNDEFINED;
+                info[j].image.sampler = nullptr;
```

Comment:
the samplers are not flexible, there's nothing you can change in them on the user side that would make sense with VT (can't change wrapping, can't change LoD parameters, can't change anisotropic mode, can't change trilinear)

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/474/files#r417486906